x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -0,0 +1,250 @@
+[role="xpack"]
+[[jwt-realm]]
+=== JWT Authentication
+
+JWT realms allow reuse of certain `identification tokens` for {es} authentication and authorization.
+JWT realms are only intended to use with JWTs from existing, external workflows.
+The format of those JWTs must be similar to Open ID Connect (OIDC) ID Tokens.
+See <<external-workflow-requirements>>.
+
+TIP: If the external workflow does not exist, use `api` tokens instead.
+
+TIP: If the external workflow is Open ID Connect, and {es} is the OIDC RP, use `oidc` realm instead.
+
+[[external-workflow-requirements]]
+==== External Workflow Requirements
+
+Consider workflows that are completely separate from {es}.
+They can be Open ID Connect, bespoke, or other, as long as they use JWS JWTs for an `identification token`.
+In those workflows, end-users interact with a third-party application, and authenticate via a third-party authentication application.
+
+Different tokens are created within those workflows for authentication, authorization, and identification.
+Reusing the authentication or authorization tokens outside of those workflows may not be desirable.
+However, reusing the `identification token` is possible.
+
+JWT realm allows reusing the `identification token` for {es} authentication and authorization.
+JWT format is similar to Open ID Connect (OIDC) ID Tokens, with some differences. See <<jwt-validation>>.
+
+[[jwt-authentication-requirements]]
+==== JWT Authentication Requirements
+
+Two credentials are required by JWT realms, a client credential and an end-user identity token.
+Both must be validated separately, because the external client and end-user are separate entities.
+
+See <<client-authentication>> and <<jwt-validation>>.
+
+[[client-authentication]]
+==== Client authentication
+
+Client authentication is one of two credentials required by JWT realms.
+See <<jwt-validation>> for the other credential.
+
+JWT realms support two client authentication settings: `shared_secret` and `none`.
+
+IMPORTANT: Client authentication is enabled by default in JWT realms.
+
+If client authentication type is `shared_secret`, JWT realm requires a request header `ES-Client-Authentication`.
+The value must be a case-insensitive match to the realm's secure string setting `client_authentication.shared_secret`.
+See <<ref-jwt-settings>>.
+
+If client authentication type is `none`, the request header `ES-Client-Authentication` is ignored.
+
+WARNING: Client authentication can be disabled, but it is not recommended.
+
+[[jwt-validation]]
+==== JWT validation
```

Comment:
I need to drop off here.

For the JWT validation section, I think we need to make a choice as to whether we want pure reference docs such as

- The `aud` value from the JWT must be an exact case-sensitive match to the realm setting `allowed_audiences`

or more generic statements such as 

- the JWT will be validated according to the JWT and OIDC specifications, in accordance with the configured <<ref-jwt-settings>>.

I think I prefer the former, and it looks like you've gone in that direction as well. My inclination is that if we're going to do that we should lean into it, and just list the validation behaviour in that reference style, rather than trying to write descriptive paragraphs.

Reply:
I made some edits. Have a look to see if they are OK.